src/CodeStyle/Core/CodeFixes/DocumentBasedFixAllProvider.cs

```diff
@@ -0,0 +1,129 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System.Collections.Generic;
+using System.Collections.Immutable;
+using System.Linq;
+using System.Threading.Tasks;
+using Microsoft.CodeAnalysis.CodeActions;
+using Microsoft.CodeAnalysis.CodeFixes;
+
+namespace Microsoft.CodeAnalysis.CodeStyle
+{
+    /// <summary>
+    /// Provides a base class to write a <see cref="FixAllProvider"/> that fixes documents independently.
+    /// </summary>
+    internal abstract class DocumentBasedFixAllProvider : FixAllProvider
+    {
+        protected abstract string CodeActionTitle
+        {
+            get;
+        }
```

Comment:
nit: nowrap.

Reply:
I'll keep this one (file is currently consistent)